linetools/isgm/utils.py

```diff
@@ -362,3 +366,138 @@ def synthesize_components(components, zcomp=None, vbuff=0*u.km/u.s):
+
+    # figure out the lowest of the chunks
+    # and sort them such that chunk1 is by definition
+    # the one with the lowest limit
+    if np.min(chunk1) <= np.min(chunk2):
+        pass
+    else: # invert them if necessary
+        aux = chunk1
+        chunk1 = chunk2
+        chunk2 = aux
+
+    # now the chunk1 is the one with the lowest value
+    # then, the only way they do not overlap is:
+    if np.min(chunk2) > np.max(chunk1):
+        return False
+    else:  # overlap exists
+        return True
+
+def overlapping_components(comp1, comp2, tol=0.2*u.arcsec):
+    """Whether two components overlap in wavelength
+    space and (ra,dec) sky position. This is useful to identify
+    components that may need to be fit together.
+
+    Parameters
+    ----------
+    comp1 : AbsComponent
+        A given AbsComponent object
+    comp2 : AbsComponent
+        A given AbsComponent object
+    tol : Quantity, optional
+        Tolerance for checking whether the two components are
+        in the same sky region. Default is 0.2*u.arcsec
+
+    Returns
+    -------
+    Boolean : True if there is overlapping wavelength range
+    and radec coordinates, otherwise False.
+    """
+
+    if not isinstance(comp1, AbsComponent):
+        raise ValueError('comp1 must be AbsComponent object.')
+    if not isinstance(comp2, AbsComponent):
+        raise ValueError('comp1 must be AbsComponent object.')
+
+    # Check whether they are in the same sky region
+    if comp1.coord.separation(comp2.coord) > tol:
+        return False
+
+    # Define wavelength chunks, appending them from each absline
+    # These will correpond to (wvmin, wvmax) tuples for each absline
+    # in the respective component
+    wobs_chunks_1 = []
+    wobs_chunks_2 = []
+    for absline in comp1._abslines:
+        cond = absline.analy['vlim'] != init_analy['vlim']  # default value?
+        if np.sum(cond) > 0:  # i.e. absline vlim not equal than the default
+            dzlim = give_dz(absline.analy['vlim'], comp1.zcomp)
+        else:  # use the vlim from component otherwise
+            dzlim = give_dz(comp1.vlim, comp1.zcomp)
+        wrest_aux = absline.wrest.to('AA').value  # in AA
+        wobs_aux = wrest_aux * (1 + comp1.zcomp + dzlim)
+        wobs_chunks_1 += [tuple(wobs_aux)]
+
+    for absline in comp2._abslines:
+        cond = absline.analy['vlim'] != init_analy['vlim']  # default value
+        if np.sum(cond) > 0:  # i.e. absline vlim not equal than the default
+            dzlim = give_dz(absline.analy['vlim'], comp2.zcomp)
+        else:  # use the vlim from component otherwise
+            dzlim = give_dz(comp2.vlim, comp2.zcomp)
+        wrest_aux = absline.wrest.to('AA').value  # in AA
+        wobs_aux = wrest_aux * (1 + comp2.zcomp + dzlim)
+        wobs_chunks_2 += [tuple(wobs_aux)]
+
+    # now we have two lists of wobs chunks:
+    # so lets do the checking until at least 1
+    # overlap is found
+    for ii in range(len(wobs_chunks_1)):
```

Comment:
Ugly for loop.

If you are just going to do this then I do think you should reduce it to a
coincident_absline method and simply loop over the abs_lines as you are doing here instead of chunks.

Yes, I am a pain.